implants/golemv2/src/agent.rs

```diff
@@ -0,0 +1,119 @@
+extern crate alloc;
+
+use alloc::collections::{BTreeMap, BTreeSet};
+use alloc::sync::Arc;
+use eldritch_libagent::agent::Agent;
+use std::sync::RwLock;
+
+#[derive(Clone)]
+pub struct GolemAgent {
+    config: Arc<RwLock<BTreeMap<String, String>>>,
+}
+
+impl GolemAgent {
+    pub fn new() -> Self {
+        let mut map = BTreeMap::new();
+        map.insert("interval".to_string(), "5".to_string());
+        Self {
+            config: Arc::new(RwLock::new(map)),
+        }
+    }
+}
+
+impl Agent for GolemAgent {
+    fn get_config(&self) -> Result<BTreeMap<String, String>, String> {
+        Ok(self.config.read().unwrap().clone())
```

Comment:
Unwrap is sus do we want the beacon to panic if it can't read the config?

If not we can .map_err(|e| e.to_string())? To bubble the error as a string.

Reply:
Completely switch to fake agent